stdlib/public/Concurrency/TaskGroup.swift

```diff
@@ -104,60 +97,54 @@ public func withTaskGroup<ChildTaskResult: Sendable, GroupResult>(
   #endif
 }
 
-/// Starts a new throwing task group which provides a scope in which a dynamic 
-/// number of tasks may be spawned.
-///
-/// Tasks added to the group by `group.spawn()` will automatically be awaited on
-/// when the scope exits. If the group exits by throwing, all added tasks will
-/// be cancelled and their results discarded.
+/// Starts a new scope in which a dynamic number of throwing tasks can be spawned.
 ///
-/// ### Implicit awaiting
-/// When the group returns it will implicitly await for all spawned tasks to
-/// complete. The tasks are only cancelled if `cancelAll()` was invoked before
-/// returning, the groups' task was cancelled, or the group body has thrown.
+/// When the group returns,
+/// it implicitly waits for all spawned tasks to complete.
+/// The tasks are canceled only if `cancelAll()` was invoked before returning,
+/// if the groups' task was canceled,
+/// or if the group body throws an error.
 ///
-/// When results of tasks added to the group need to be collected, one can
-/// gather their results using the following pattern:
+/// To collect the results of tasks that were added to the group,
+/// use the following pattern:
 ///
-///     while let result = await try group.next() {
-///       // some accumulation logic (e.g. sum += result)
+///     while let result = await group.next() {
+///         // some accumulation logic (e.g. sum += result)
 ///     }
 ///
 /// It is also possible to collect results from the group by using its
 /// `AsyncSequence` conformance, which enables its use in an asynchronous for-loop,
 /// like this:
 ///
-///     for try await result in group {
-///       // some accumulation logic (e.g. sum += result)
+///     for await result in group {
+///         // some accumulation logic (e.g. sum += result)
 ///      }
+/// ◊TR: Which of the above do we prefer?  Why give two recommendations?
+///
+/// When tasks are added to the group
+/// using the `Group.spawn(priority:operation:)` method,
+/// they might begin executing immediately.
+/// Even if their results are not collected explicitly and such task throws,
+/// and was not yet canceled,
+/// it may result in the `withTaskGroup` throwing.
+/// ◊TR: What does the above "such task throws" mean?
```

Comment:
Let's remove that wording; we're not doing it.

It used to be about:

```
try await withThrowingTaskGroup(...) { g in
g.spawn { throw Boom() } 
}
```

so we never "collected" that boom; in practice we will ignore that error; We could have also decided to re-throw it, but that's not what we do nowadays